### PR TITLE
Fix missing auth header did not lead to 401

### PR DESCRIPTION
**Pull request by KochTobi**
Changed auth configuration to match the new endpoint.
A missing auth header at the token generation endpoint now leads to 401.

### CODE REVIEW

**Review by sven1103 — APPROVED** [review]
🚀

File: src/test/java/life/qbic/subscriptions/SubscriptionControllerTest.java [hunk 1]
```diff
@@ -85,21 +84,15 @@ void whenValidInputIsProvidedPostSubscriptionsTokensRespondsCreated() throws Exc
         .andExpect(content().string(encrypted));
   }
 
-  @Disabled("Disabled until authorization is tackled")
   @Test
-  @DisplayName("When authorization is missing, DELETE /subscriptions responds UNAUTHORIZED")
+  @DisplayName("When authorization is missing, DELETE /subscriptions responds NO CONTENT")
```
Comment (Steffengreiner) [inline src/test/java/life/qbic/subscriptions/SubscriptionControllerTest.java:88]: The test name and respond code don't fit. 
Why was the response changed to "no Content". 
I'd expect a 401 or 403 API response if the client is unauthorized?

Comment (Steffengreiner) [reply src/test/java/life/qbic/subscriptions/SubscriptionControllerTest.java:88]: Also the post and delete requests should respond with the same api respond if the client is unauthorized

**Review by Steffengreiner — CHANGES_REQUESTED** [review]
Nice changes concerning the configuration, however I'm confused on the change in the API responds codes

File: src/test/java/life/qbic/subscriptions/SubscriptionControllerTest.java (hunk 1 above)
Comment (KochTobi) [reply src/test/java/life/qbic/subscriptions/SubscriptionControllerTest.java:88]: Hi Steffen, the POST endpoint for token generation should require basic auth. However, the subscription cancellation via token does not (as the possession of the token itself is enough)

**Pull request merged**